expect/_asymmetric_matchers.ts

```diff
@@ -69,11 +70,78 @@ export class ArrayContaining extends AsymmetricMatcher<any[]> {
   constructor(arr: any[]) {
     super(arr);
   }
+
   equals(other: any[]): boolean {
-    return this.value.every((e) => other.includes(e));
+    return Array.isArray(other) && this.value.every((e) => other.includes(e));
   }
 }
 
 export function arrayContaining(c: any[]): ArrayContaining {
   return new ArrayContaining(c);
 }
+
+export class CloseTo extends AsymmetricMatcher<number> {
+  readonly #precision: number;
+
+  constructor(num: number, precision: number = 2) {
+    super(num);
+    this.#precision = precision;
+  }
+
+  equals(other: number): boolean {
+    if (typeof other !== "number") {
+      return false;
+    }
+
+    if (
+      (this.value === Number.POSITIVE_INFINITY &&
+        other === Number.POSITIVE_INFINITY) ||
+      (this.value === Number.NEGATIVE_INFINITY &&
+        other === Number.NEGATIVE_INFINITY)
+    ) {
+      return true;
+    }
+
+    return Math.abs(this.value - other) < Math.pow(10, -this.#precision) / 2;
+  }
+}
+
+export function closeTo(num: number, numDigits?: number) {
+  return new CloseTo(num, numDigits);
+}
+
+export class StringContaining extends AsymmetricMatcher<string> {
+  constructor(str: string) {
+    super(str);
+  }
+
+  equals(other: string): boolean {
+    if (typeof other !== "string") {
+      return false;
+    }
+
+    return other.includes(this.value);
+  }
+}
+
+export function stringContaining(str: string) {
```

Comment:
Please don't forget to define explicit return types (except for `void` or `Promise<void>`). Ditto elsewhere.